sc-memory/sc-core/sc_memory_context_manager.c

```diff
@@ -37,102 +46,310 @@ struct _sc_event_emit_params
+  sc_iterator3_free(it3);
 
-  sc_memory_arc_new(
+  sc_addr const auth_arc_addr = sc_memory_arc_new(
       s_memory_default_ctx, sc_type_arc_pos_const_temp, manager->concept_authenticated_user_addr, user_addr);
+  _sc_context_set_access_levels_for_element(auth_arc_addr, SC_CONTEXT_ACCESS_LEVEL_TO_ALL_ACCESS_LEVELS);
 
   return SC_RESULT_OK;
 }
 
 /*! Function that handles the removal of authentication for a user and its associated memory context.
- * @param event Pointer to the sc_event triggering the context removal.
- * @param arc_addr sc-address representing removed sc-arc to the authenticated user..
+ * @param event Pointer to the sc-event triggering the context removal.
+ * @param initiator_addr sc-address representing user that initiated this request.
+ * @param connector_addr sc-address representing removed sc-connector to the authenticated user.
+ * @param connector_type sc-type of created sc-connector to the user.
  * @param user_addr sc-address representing the user whose authentication is being revoked.
  * @returns Returns a result code indicating the success or failure of the operation (SC_RESULT_OK on success).
  * @note This function is called in response to a sc-event and is responsible for removing authentication
  * for user and its associated memory context.
  */
 sc_result _sc_memory_context_manager_on_unauthentication_request_user(
     sc_event const * event,
-    sc_addr arc_addr,
+    sc_addr initiator_addr,
+    sc_addr connector_addr,
+    sc_type connector_type,
     sc_addr user_addr)
 {
-  sc_unused(&arc_addr);
+  sc_unused(&initiator_addr);
+  sc_unused(&connector_addr);
+
+  // Only temporal sc-arcs can be used
+  if (sc_type_check(connector_type, sc_type_arc_pos_const_temp) == SC_FALSE)
+    return SC_RESULT_OK;
 
   sc_memory_context_manager * manager = sc_event_get_data(event);
   sc_memory_context * ctx = _sc_memory_context_get_impl(manager, user_addr);
   if (ctx == null_ptr)
-  {
     _sc_context_remove_user_access_levels(user_addr, SC_CONTEXT_ACCESS_LEVEL_AUTHENTICATED);
+  else
+    sc_context_remove_context_access_levels(ctx, SC_CONTEXT_ACCESS_LEVEL_AUTHENTICATED);
+
+  sc_addr const auth_arc_addr = sc_memory_arc_new(
+      s_memory_default_ctx, sc_type_arc_neg_const_temp, manager->concept_authenticated_user_addr, user_addr);
+  _sc_context_set_access_levels_for_element(auth_arc_addr, SC_CONTEXT_ACCESS_LEVEL_TO_ALL_ACCESS_LEVELS);
+
+  return SC_RESULT_OK;
+}
+
+/**
+ * @brief Adds basic action class access levels to the sc-memory context manager.
+ * @param _action_class_system_idtf System identifier of the action class.
+ * @param _access_levels Access levels to be added for the action class.
+ */
+#define sc_context_manager_add_basic_action_class_access_levels(_action_class_system_idtf, _access_levels) \
+  ({ \
+    sc_addr _action_class_addr; \
+    sc_helper_resolve_system_identifier(s_memory_default_ctx, _action_class_system_idtf, &_action_class_addr); \
```

Comment:
Default access classes will be created without class type

Reply:
I think sc-machine doesn't respond for this. Types can be different. As for me, I think that 
1. simple set types are more appropriate than class types for nodes from which temporal arcs are outgoing
2. semantic types can removed from sc-machine in the future